plugins/plugin_utils/json_to_xml.py

```diff
@@ -0,0 +1,45 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2021 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+"""
+The json_to_xml plugin
+"""
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import ast
+from ansible.errors import AnsibleError
+
+
+def _raise_error(msg):
+    """Raise an error message, prepend with filter name
+
+    :param msg: The message
+    :type msg: str
+    :raises: AnsibleError
+    """
+    error = "Error when using plugin 'json_to_xml': {msg}".format(msg=msg)
+    raise AnsibleError(error)
+
+
+def json_to_xml(data, engine):
+    """Convert data which is in json to xml"
+
+    :param data: The data passed in (data|json_to_xml(...))
+    :type data: xml
+    :param engine: Conversion library default=xmltodict
+    """
+    if engine == "xmltodict":
+        import xmltodict
+
+        try:
+            data = ast.literal_eval(data)
```

Comment:
why is literal_eval required here?

Reply:
This is needed as we are getting data as string with single quoted key and value pair like this  {'interface-configurations': {'@xmlns': 'http://cisco.com/ns/yang/Cisco-IOS-XR-ifmgr-cfg', 'interface-configuration': None}}. if we directly pass this to xmltodict its throwing "json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)"